node/silkworm/db/buffer.cpp

```diff
@@ -96,19 +101,188 @@ void Buffer::update_storage(const evmc::address& address, uint64_t incarnation,
     }
     if (block_number_ >= prune_from_) {
         changed_storage_.insert(address);
-        ByteView change_val{zeroless_view(initial)};
-        if (block_storage_changes_[block_number_][address][incarnation].insert_or_assign(location, change_val).second) {
-            bump_batch_size(8 + kPlainStoragePrefixLength, kHashLength + change_val.size());
+        ByteView initial_val{zeroless_view(initial)};
+        if (block_storage_changes_[block_number_][address][incarnation]
+                .insert_or_assign(location, initial_val)
+                .second) {
+            batch_history_size_ += kPlainStoragePrefixLength + kHashLength + initial_val.size();
         }
     }
 
     if (storage_[address][incarnation].insert_or_assign(location, current).second) {
-        bump_batch_size(kPlainStoragePrefixLength, kHashLength + zeroless_view(current).size());
+        batch_state_size_ += kPlainStoragePrefixLength + kHashLength + kHashLength;
```

Comment:
`kHashLength` instead of `zeroless_view(current).size())` overestimates batch size. A lot of storage values are relatively small numbers, nowhere close to 2^256.

Reply:
I don't agree ... we're measuring the occupation in RAM of `storage_` which is not zerolessed until is flushed to database.
Actual RAM occupation is its full size, not its zerolessed view size.

Reply:
BTW ... this comment made me realize was doing the wrong accounting on in memory `accounts_`
Thanks for heads up